### DISCUSSION TITLE
How is the syntax sugar implemented ?

### DISCUSSION THREAD

**Discussion by userrand** [WLJSTeam/wljs-notebook#23] [General]
I was thinking about how I could visually replace lengthy function names like 

`MultivariateHypergeometricDistribution`

 without having to redefine a function each time I use it in a session. Like that, under the hood, it's still the same long name but visually I see it as MuHyDi or something and I have nothing to redefine. I could also use an autocomplete/text expansion for the reverse `MuHyDi->MultivariateHypergeometricDistribution`. Then I realized that this code does that with its syntactic sugar. 

My question is how is the syntactic sugar in this code made? Would you happen to know if it's the same idea as rendering latex in obsidian for example? If that's the case then I suppose I could not implement something at the level of the OS where visually on my screen I see something different then the characters that are on the application.

**Comment by KirillBelovTest** [comment]
It seems like an interesting idea to me, too. In mathematica are **[Interpretation](http://reference.wolfram.com/language/ref/Interpretation.html)**.

**Comment by JerryI** [comment]
Sure it is definitely possible.
There are many ways how it can be implemented. In general, if you want to lear more about how decorations work, please see this [repo](https://github.com/JerryI/codemirror6-mathematica-sugar) (# Run an demo).

To generalize all of this I am currently using `FrontEndBox[exp, disp]` abstraction, where `disp` contains any code that will be displayed instead of the `expr`.  Now all Mathematica's `Boxes`, `InterpretationBox` are translated into this (available only in `dev` branch). I am thinking about to implement full features for [Interpretation](http://reference.wolfram.com/language/ref/Interpretation.html) mentioned by @KirillBelovTest, so that you could use native Mathematica's approach to this problem.

**Reply by userrand** [reply]
thanks for the link  :+1:

**Comment by JerryI** [comment]
I am currently busy with [wljs-interpreter](https://github.com/JerryI/wljs-interpreter) and docs for it, since there is a conference on Wolfram Language ;) But I will definitely come back to frontend project and finally upload the recent version with plugins support (which is almost ready to go, no `nodejs` installed or recompilation needed. As you suggested a long time ago with a proper plugin management system).

**Reply by userrand** [reply]
very nice ! sounds great (: :+1:

**Comment by JerryI** [comment]
Finally was implemented #27 

```mathematica
AVeryLongFunction /: MakeBoxes[AVeryLongFunction, opts___] := InterpretationBox[Ashort, AVeryLongFunction]

AVeryLongFunction[123]
```
